src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5429,9 +5438,28 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
 
         public override BoundNode VisitThrowExpression(BoundThrowExpression node)
         {
-            var result = base.VisitThrowExpression(node);
-            SetUnknownResultNullability();
-            return result;
+            VisitThrow(node.Expression);
+            return null;
+        }
+
+        public override BoundNode VisitThrowStatement(BoundThrowStatement node)
+        {
+            VisitThrow(node.ExpressionOpt);
+            return null;
+        }
+
+        private void VisitThrow(BoundExpression expr)
+        {
+            var result = VisitRvalueWithResult(expr);
+            // We treat `throw null` like an explicit `throw new System.NullReferenceException()`. The user is being explicit enough, so a warning would be annoying.
+            if (!expr.IsLiteralNull())
+            {
+                if (result.NullableAnnotation.IsAnyNullable())
```

Comment:
>result.NullableAnnotation.IsAnyNullable() [](start = 20, length = 41)

I don't think this check will properly handle an unconstrained type parameter. Please add a test for a type parameter constrained to ```Exception?```.

Reply:
I still need to add this

Reply:
Done now. Thanks